sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DistroHelper.java

```diff
@@ -165,39 +164,61 @@ public void savePropertiesToServer(DistroProperties properties, Server server) t
 	 */
 	public static Artifact parseDistroArtifact(String distro, VersionsHelper versionsHelper) throws MojoExecutionException {
 		String[] split = distro.split(":");
-
 		if (split.length > 3) {
 			throw new MojoExecutionException("Invalid distro: " + distro);
 		}
-
 		String groupId = split.length == 3 ? split[0] : Artifact.GROUP_DISTRO;
 		String artifactId = split[split.length - 2];
 		String version = split[split.length - 1];
+		return normalizeArtifact(new Artifact(artifactId, version, groupId), versionsHelper);
+	}
 
-		if (versionsHelper != null && version.contains(SDKConstants.LATEST_VERSION_BATCH_KEYWORD)) {
-			if (version.equalsIgnoreCase(SDKConstants.LATEST_SNAPSHOT_BATCH_KEYWORD)) {
-				version = versionsHelper.getLatestSnapshotVersion(new Artifact(artifactId, version, groupId));
-			} else if (version.equalsIgnoreCase((SDKConstants.LATEST_VERSION_BATCH_KEYWORD))) {
-				version = versionsHelper.getLatestReleasedVersion(new Artifact(artifactId, version, groupId));
```

Comment:
There is a lot of scattered code in the SDK that does "special purpose stuff".  One example of that is resolving "referenceapplication" to either "referenceapplication-package" or "referenceapplication-distro" depending on the version.  This was not being applied everywhere equally, so this is an effort to create a method for doing this that can be called from various places to gain consistency.  More refactoring and consolidation down the road will help, but this is a start.